### PR TITLE
Simplify and clean up co2 calculation code

### PR DESCRIPTION
**Pull request by yichengt900**
This PR addresses issue #71. The following modifications have been made:

1. Renamed `FMS_ocmip2_co2calc.F90` to `FMS_ocmip2_co2calc.F90.archive`.
2. Created `FMS_co2calc.F90`, which contains the necessary routines/interface for `mocsy` co2 calculation.
3.  Simplified `generic_COBALT.F90`, `generic_BLING.F90`, and `generic_abiotic.F90`, so they now work properly with the new `FMS_co2calc.F90`.

The only thing I have not removed yet is the `co2_calc` option in the namelist. Removing `co2_calc` would affect the model configurations because we would have to take it out of `input.nml` to prevent the model from complaining about an unknown namelist.

This PR should not change answers and it has passed both the 1D case. and `NWA12-COBALT` RT  case.

### CODE REVIEW

**Comment by charliestock** [comment]
This is great @yichengt900   Do you think a 1 year NWA12 test that looks at the DIC and/or air-sea exchange before and after this is needed to ensure that no subtle change has been introduced?

**Comment by yichengt900** [comment]
@charliestock, good point. I conducted a one-year NWA12 test to check if the DIC results from this PR are identical to the results from the current `dev/cefi` branch. The good news is they are identical (see the plots below). I think this PR is now ready for review.

`ALK, DIC, omega from current `dev/cefi` branch`:
alk_dic_omega

`ALK, DIC, omega from this PR`:
alk_dic_omega

**Review by charliestock — APPROVED** [review]
Hello @yichengt900.  Thank you for cleaning this up.  I think it will help us avoid many issues moving forward.  I have checked the changes and they all seemed sensible.  The fact that the simulated DIC and alkalinity fields were bitwise reproducible even with a full year also provides high confidence.  I have approved the request, but had 2 thoughts to consider before finalizing (or for a later pull request):

1) I will run the details by John Dunne offline.
2) Are we sure that FMS_ocmip2_co2calc.F90 isn't called anywhere else?  For example, in the air-sea exchange codes?

For the latter, Eric Stofferahn mentioned that there are a number of "ocmip2" linked codes that Brandon Reichl was modifying for his work.  It sounded like these were unrelated to the CO2 calculation that we are working on here, but could we double check that there are no cross-linkages?

Thanks again!
Charlie

**Comment by yichengt900** [comment]
@charliestock,

Thank you. Regarding your 2 question, I checked the FMS air-sea exchange codes and found some ocmip2-linked codes. However, none of them seem to be related to `FMS_ocmip2_co2calc.F90` directly. I will keep this PR open for a while to ensure everything is okay before we merge it.

File: generic_tracers/generic_COBALT.F90 [hunk 1]
```diff
@@ -271,13 +265,12 @@ subroutine generic_COBALT_register(tracer_list)
       write (stdoutunit,*) trim(note_header), 'Simulating radiocarbon'
     endif
 
-    if (trim(co2_calc) == 'ocmip2') then
-      write (stdoutunit,*) trim(note_header), 'Using FMS OCMIP2 CO2 routine'
-    else if (trim(co2_calc) == 'mocsy') then
+    if (trim(co2_calc) == 'mocsy') then
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90:268]: Do we need this moving forward, if we're not providing multiple options for the co2 calculation?

**Review by jessluo — APPROVED** [review]
This looks good to me. I just left one quick question but otherwise no additional suggestions for this PR.

File: generic_tracers/generic_COBALT.F90 (hunk 1 above)
Comment (jessluo) [reply generic_tracers/generic_COBALT.F90:268]: Oh, I see there is already an issue open regarding this: [https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/issues/71](url)

Comment (yichengt900) [reply generic_tracers/generic_COBALT.F90:268]: @jessluo, thank you! Yes, this is the first step for #71, simplifying the current CO2 calculation module. We are keeping `co2_calc` in this PR but will consider removing it in the future if we offer only one option to users.

**Pull request merged**